supertokens_python/utils.py

```diff
@@ -299,8 +299,16 @@ def get_top_level_domain_for_same_site_resolution(url: str) -> str:
 
     if hostname.startswith("localhost") or is_an_ip_address(hostname):
         return "localhost"
+
     parsed_url: Any = extract(hostname, include_psl_private_domains=True)
     if parsed_url.domain == "":  # type: ignore
+        if hostname.endswith(".amazonaws.com"):
+            # Example: url http://ec2-xx-yyy-zzz-0.compute-1.amazonaws.com
+            # return ec2-xx-yyy-zzz-0.compute-1.amazonaws.com
+            # If user deploys the website on the same ec2 instance as the API
+            # this use SameSite=lax, otherwise SameSite=none
+            return hostname
```

Comment:
- return just amazonaws.com
- modify the place where you are checking against the public suffix list to check for this domain too
- add tests

Reply:
done